src/Config/RuntimeConfigLoader.cs

```diff
@@ -0,0 +1,312 @@
+        _baseConfigFileName = baseConfigFileName;
+        _connectionString = connectionString;
+    }
+
+    /// <summary>
+    /// Load the runtime config from the specified path.
+    /// </summary>
+    /// <param name="path">The path to the dab-config.json file.</param>
+    /// <param name="config">The loaded <c>RuntimeConfig</c>, or null if none was loaded.</param>
+    /// <returns>True if the config was loaded, otherwise false.</returns>
+    public bool TryLoadConfig(string path, [NotNullWhen(true)] out RuntimeConfig? config)
+    {
+        if (_fileSystem.File.Exists(path))
+        {
+            string json = _fileSystem.File.ReadAllText(path);
+            return TryParseConfig(json, out config, connectionString: _connectionString);
+        }
+
+        config = null;
+        return false;
+    }
+
+    /// <summary>
+    /// Parses a JSON string into a <c>RuntimeConfig</c> object
+    /// </summary>
+    /// <param name="json">JSON that represents the config file.</param>
+    /// <param name="config">The parsed config, or null if it parsed unsuccessfully.</param>
+    /// <returns>True if the config was parsed, otherwise false.</returns>
+    public static bool TryParseConfig(string json, [NotNullWhen(true)] out RuntimeConfig? config, ILogger? logger = null, string? connectionString = null)
+    {
+        JsonSerializerOptions options = GetSerializationOptions();
+
+        try
+        {
+            config = JsonSerializer.Deserialize<RuntimeConfig>(json, options);
+
+            if (config is null)
+            {
+                return false;
+            }
+
+            if (!string.IsNullOrEmpty(connectionString))
+            {
+                config = config with { DataSource = config.DataSource with { ConnectionString = connectionString } };
+            }
+        }
+        catch (JsonException ex)
+        {
+            string errorMessage = $"Deserialization of the configuration file failed.\n" +
+                        $"Message:\n {ex.Message}\n" +
+                        $"Stack Trace:\n {ex.StackTrace}";
+
+            if (logger is null)
+            {
+                // logger can be null when called from CLI
+                Console.Error.WriteLine(errorMessage);
+            }
+            else
+            {
+                logger.LogError(ex, errorMessage);
+            }
+
+            config = null;
+            return false;
+        }
+
+        return true;
+    }
+
+    public static JsonSerializerOptions GetSerializationOptions()
+    {
+        JsonSerializerOptions options = new()
+        {
+            PropertyNameCaseInsensitive = false,
+            PropertyNamingPolicy = new HyphenatedNamingPolicy(),
+            ReadCommentHandling = JsonCommentHandling.Skip,
+            WriteIndented = true,
+        };
+        options.Converters.Add(new HyphenatedJsonEnumConverterFactory());
+        options.Converters.Add(new RestRuntimeOptionsConverterFactory());
```

Comment:
Class summaries for all of these converters would make it a lot easier to understand the intent behind using the converter.

Reply:
I've added some to the ones that are potentially ambigious, such as the `StringJsonConverterFactory`, but the ones that are named for the type, such as `RestRuntimeOptionsConverterFactory` being for `RestRuntimeOptions`, will class summaries provide much additional value?

Reply:
hmm makes sense.